src/main/java/com/atlassian/db/replica/api/reason/Reason.java

```diff
@@ -55,6 +63,35 @@ public String toString() {
         return "Reason{" +
             "name='" + name + '\'' +
             ", isRunOnMain=" + isRunOnMain +
+            ", isWrite=" + isWrite +
             '}';
     }
+
+    private static class ReasonBuilder {
+        private String name;
+        private boolean isRunOnMain;
+        private boolean isWrite;
+
+        private ReasonBuilder(final String name) {
+            this.name = name;
+        }
+
+        static ReasonBuilder name(final String name) {
+            return new ReasonBuilder(name);
```

Comment:
Setting name overrides other fields!

ReasonBulder("a").name("b").isWrite(true) != ReasonBulder("a").isWrite(true).name("b")

Reply:
I see it's static, so the name can't be changed. Let's make it `final`.

Reply:
fixed